src/main/java/gov/nysenate/openleg/controller/api/apiuser/ApiUserEmailSubscriptionCtrl.java

```diff
@@ -0,0 +1,57 @@
+package gov.nysenate.openleg.controller.api.apiuser;
+
+import gov.nysenate.openleg.controller.api.base.BaseCtrl;
+import gov.nysenate.openleg.dao.auth.SqlApiUserDao;
+import gov.nysenate.openleg.model.auth.ApiUser;
+import gov.nysenate.openleg.model.auth.ApiUserSubscriptionType;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.*;
+
+@RestController
+@RequestMapping(BaseCtrl.BASE_API_PATH + "/email/subscription")
+public class ApiUserEmailSubscriptionCtrl extends BaseCtrl {
+
+    @Autowired
+    protected SqlApiUserDao sqlApiUserDao;
+
+    @RequestMapping(value = "/update", method = RequestMethod.POST)
+    public void updateSubscriptions(@RequestParam String key, @RequestBody List<String> body) {
+        Set<ApiUserSubscriptionType> subscriptions = new HashSet<>();
+        for (String sub : body) {
+            subscriptions.add(ApiUserSubscriptionType.valueOf(sub));
```

Comment:
We have a built in method for enum validation that handles cases where invalid values are passed in.
See BaseCtrl#getEnumParameter